src/negotiation/taker/negotiator.ts

```diff
@@ -67,22 +77,28 @@ export class Negotiator {
    * opportunity to the lib consumer to know that this maker returns invalid orders and the details of such order.
    * @param criteria - The criteria of the order to be requested from the maker.
    */
-  public async getOrder(criteria: MatchingCriteria): Promise<Order> {
+  public async getOrder(criteria: MatchingCriteria): Promise<TakerOrder> {
     const tradingPair = matchingCriteriaToTradingPair(criteria);
     const rawOrder = await this.makerClient.getOrderByTradingPair(tradingPair);
 
-    return new Order(rawOrder, criteria, this.execAndTakeOrder.bind(this));
+    return new TakerOrder(rawOrder, criteria, this.execAndTakeOrder.bind(this));
   }
 
-  private async execAndTakeOrder(
-    rawOrder: RawOrder
-  ): Promise<Swap | undefined> {
-    const executionParams = await this.makerClient.getExecutionParams(rawOrder);
+  /**
+   * Take the {@link Order} from the maker and initiate the {@link Swap} execution.
```

Comment:
A lib consumer should not use this function but use `TakerOrder.take()` instead so either don't document it or just to use `TakerOrder.take()`.

Reply:
Yeah, it was private before, but I had problems referencing it in a `{@link }` then 😓
Maybe we should just not reference it, and make it private again.

Reply:
```
declare class Order {
    readonly rawOrder: RawOrder;
    readonly criteria: MatchingCriteria;
    readonly takeOrder: (rawOrder: RawOrder) => Promise<Swap | undefined>;
    /**
     * **Note: This should not be used, `Order` should be created by using {@link getOrder}
     * @param rawOrder - The parameters of the order, as received from the maker.
     * @param criteria - The criteria used to filter/retrieve this order.
     * @param takeOrder - Function that notifies the maker to take the order and initiates the swap, see {@link execAndTakeOrder}.
     */
    constructor(rawOrder: RawOrder, criteria: MatchingCriteria, takeOrder: (rawOrder: RawOrder) => Promise<Swap | undefined>);
    /**
     * @description: Return whether an order matches the passed criteria.
     */
```

It is mentioned in the comment here, and I could not reference it properly being private...

Reply:
I'll make it private again

Reply:
private again, removed reference in comment and comment on the private function.